app/src/features/observations/BlockObservationPage.tsx

```diff
@@ -140,6 +160,41 @@ const BlockObservationPage = () => {
     return <CircularProgress className="pageProgress" size={40} />;
   }
 
+  const handleSaveAndExit = async (values: ICreateBlockObservationPostRequest) => {
+    if (!formikRef?.current) {
+      return;
+    }
+
```

Comment:
here we need to do a formikRef.current.handleSubmit or something to ensure form fields are all valid and no yup errors I think

Reply:
await formikRef.current?.submitForm();

Reply:
```
await formikRef.current?.submitForm();

    const isValid = await validateFormFieldsAndReportCompletion(
      formikRef.current?.values,
      formikRef.current?.validateForm
    );

    if (!isValid) {
      showCreateErrorDialog({
        dialogTitle: 'Add Observation Form Incomplete',
        dialogText:
          'The form is missing some required fields/sections highlighted in red. Please fill them out and try again.'
      });

      return;
    }
```